### PR TITLE
Add ACCOUNT_CONFIRMATION_REQUESTED event

### PR DESCRIPTION
**Pull request by SzymJ**
I want to merge this change because it adds `ACCOUNT_CONFIRMATION_REQUESTED` event.

Part of #12317 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

**Review by maarcingebala — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Comment by patrys** [comment]
Since this is a new event, here are my thoughts:

1. We should always send the confirmation requested event.
2. The event should contain information if confirmation is mandatory or not.
3. Even if confirmation is not required to log in, it should be required to claim previously placed orders (and orders using the same email address placed without logging in)

**Comment by SzymJ** [comment]
> Since this is a new event, here are my thoughts:
> 
> 1. We should always send the confirmation requested event.
> 2. The event should contain information if confirmation is mandatory or not.
> 3. Even if confirmation is not required to log in, it should be required to claim previously placed orders (and orders using the same email address placed without logging in)

@patrys so:
1. should we send an event without checking `enable_account_confirmation_by_email` flag?
2. If point `1` is true then we can get this flag in the payload from `Shop` object.
3. Do you mean to add a task for matching orders with an email to this mutation?

**Comment by patrys** [comment]
1. Yes, I believe we should always generate allow email confirmation, whether it's required or not.
2. Great!
3. The mutation that confirms your account should also claim any orders placed using the same email, not the one sending the token.

**Comment by SzymJ** [comment]
@patrys it looks like `ConfirmAccount` already run matching orders with email

File: saleor/graphql/account/mutations/account.py [hunk 1]
```diff
@@ -162,12 +164,26 @@ def save(cls, info: ResolveInfo, user, cleaned_input):
...
+                params = urlencode({"email": user.email, "token": token})
+                confirm_url = prepare_url(params, cleaned_input["redirect_url"])
+
+                cls.call_event(
+                    manager.account_confirmation_requested,
+                    user,
+                    cleaned_input["channel"],
+                    token,
+                    confirm_url,
+                )
```
Comment (patrys) [inline saleor/graphql/account/mutations/account.py:186]: It's still only sent if account confirmation is required.

Comment (SzymJ) [reply saleor/graphql/account/mutations/account.py]: Done, additionally, as @fowczarek mentioned in #13170 I make notify and webhook event to use the same token.

**Review by patrys — APPROVED** [review]

File: saleor/graphql/schema.graphql [hunk 2]
```diff
@@ -1687,6 +1687,7 @@ type WebhookEvent @doc(category: "Webhooks") {
 enum WebhookEventTypeEnum @doc(category: "Webhooks") {
   """All the events."""
   ANY_EVENTS
+  ACCOUNT_CONFIRMATION_REQUESTED
```
Comment (patrys) [inline saleor/graphql/schema.graphql:1690]: Description missing!

File: saleor/graphql/schema.graphql [hunk 3]
```diff
@@ -2318,6 +2319,7 @@ type WebhookEventAsync @doc(category: "Webhooks") {
 enum WebhookEventTypeAsyncEnum @doc(category: "Webhooks") {
   """All the events."""
   ANY_EVENTS
+  ACCOUNT_CONFIRMATION_REQUESTED
```
Comment (patrys) [inline saleor/graphql/schema.graphql:2322]: Description missing!

File: saleor/graphql/schema.graphql [hunk 4]
```diff
@@ -27426,6 +27429,40 @@ enum VolumeUnitsEnum {
   ACRE_FT
 }
 
+"""
+Event sent when account confirmation requested.
+
+Added in Saleor 3.15.
+"""
+type AccountConfirmationRequested implements Event @doc(category: "Users") {
```
Comment (patrys) [inline saleor/graphql/schema.graphql:27437]: It does not include information if the confirmation is required to log in or optional.

Comment (patrys) [inline saleor/graphql/schema.graphql:27437]: It does not include information if the confirmation is required to log in or optional.

**Comment by szczecha** [comment]
Should we add a description of the event? 
Screenshot 2023-06-26 at 11 34 20

**Review by yellowee — APPROVED** [review]

**Pull request merged**